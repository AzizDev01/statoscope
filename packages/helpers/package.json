{
  "name": "@statoscope/helpers",
  "version": "4.1.2",
  "description": "Statoscope utils",
  "main": "./src/index.js",
  "scripts": {
  },
  "repository": "https://github.com/statoscope/statoscope.git",
  "author": {
    "name": "Sergey Melyukov",
    "url": "https://twitter.com/smelukov"
  },
  "files": [
    "src/*"
  ],
  "license": "MIT",
  "homepage": "https://github.com/statoscope/statoscope/",
  "devDependencies": {
  },
  "dependencies": {
  }
}
