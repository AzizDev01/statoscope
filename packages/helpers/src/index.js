module.exports.jora = require('./jora');
