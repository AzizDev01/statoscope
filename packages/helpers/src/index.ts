import jora from './jora';

export { jora };
