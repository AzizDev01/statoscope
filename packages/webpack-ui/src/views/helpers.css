.root > *:not(:last-child) {
  margin-left: 0;
  margin-right: 1ex;
}
