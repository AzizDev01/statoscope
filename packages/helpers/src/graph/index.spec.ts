import { serializeSolutionPath } from '../../../../test/helpers';
import Graph from './';

test('makeNode', () => {
  const graph = new Graph();
  const foo = graph.makeNode('foo', 'foo data');

  expect(foo.id).toBe('foo');
  expect(foo.data).toBe('foo data');
});

test('hasNode', () => {
  const graph = new Graph();
  const graph2 = new Graph();

  graph.makeNode('foo', 'foo data');
  graph2.makeNode('bar', 'bar data');

  expect(graph.hasNode('foo')).toBe(true);
  expect(graph.hasNode('bar')).toBe(false);

  expect(graph2.hasNode('foo')).toBe(false);
  expect(graph2.hasNode('bar')).toBe(true);
});

test('getNode', () => {
  const graph = new Graph();
  const foo = graph.makeNode('foo', 'foo data');
  const bar = graph.makeNode('bar', 'bar data');

  expect(graph.getNode('fff')).toBeNull();
  expect(graph.getNode('foo')).toEqual(foo);
  expect(graph.getNode('bar')).toEqual(bar);
});

test('addChild', () => {
  const graph = new Graph();
  const foo = graph.makeNode('foo', 'foo data');
  const bar = graph.makeNode('bar', 'bar data');
  const baz = graph.makeNode('baz', 'baz data');

  graph.addChild(baz, foo);
  bar.addChild(baz);

  expect([...baz.parents]).toEqual([foo, bar]);
  expect([...foo.children]).toEqual([baz]);
  expect([...bar.children]).toEqual([baz]);
});

test('findPaths', () => {
  const graph = new Graph();
  const target = graph.makeNode('target', 'target data');
  const foo = graph.makeNode('foo', 'foo data');
  const bar = graph.makeNode('bar', 'bar data');
  const baz = graph.makeNode('baz', 'baz data');

  graph.addChild(baz, foo);
  graph.addChild(baz, bar);

  graph.addChild(foo, target);

  expect(serializeSolutionPath(graph.findPaths(baz, target))).toMatchSnapshot();

  graph.addChild(bar, target);

  expect(serializeSolutionPath(graph.findPaths(baz, target))).toMatchSnapshot();
  expect(serializeSolutionPath(baz.findPathsTo(target, 1))).toMatchSnapshot();
});
