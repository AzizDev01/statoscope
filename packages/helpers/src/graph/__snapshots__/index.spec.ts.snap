// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`findPaths 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "node": Object {
            "id": "baz",
          },
        },
      ],
      "node": Object {
        "id": "foo",
      },
    },
  ],
  "node": Object {
    "id": "target",
  },
}
`;

exports[`findPaths 2`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "node": Object {
            "id": "baz",
          },
        },
      ],
      "node": Object {
        "id": "foo",
      },
    },
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "node": Object {
            "id": "baz",
          },
        },
      ],
      "node": Object {
        "id": "bar",
      },
    },
  ],
  "node": Object {
    "id": "target",
  },
}
`;

exports[`findPaths 3`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "node": Object {
            "id": "baz",
          },
        },
      ],
      "node": Object {
        "id": "foo",
      },
    },
  ],
  "node": Object {
    "id": "target",
  },
}
`;
